### ISSUE TITLE
pelican-quickstart creates incompatible fabfile.py with python 3

### ISSUE THREAD

**Issue by 3e4** [getpelican/pelican#2354]
I installed ``pelican`` 3.7.1 in an virtualenv (python=3.6). I used pip:

    pip install pelican fabric markdown

This installed fabric 2.1.3. After creating a default page with ``pelican-quickstart`` (defaults), I run ``fab build``. I get the following error trace:

    ....
      File "/tmp/pelican/fabfile.py", line 1, in <module>
        from fabric.api import *
    ModuleNotFoundError: No module named 'fabric.api'

If I install the fabric port [Fabric3](https://github.com/mathiasertl/fabric/) instead, I get an error about ``SocketServer`` that is called ``socketserver`` in Python 3.x.:

    ModuleNotFoundError: No module named 'SocketServer'

If I fix this in line 6 of the ``fabfile.py``, I can build the site.

It appears that ``pelican-quickstart`` is not properly tested with Python 3.4+? And there is no mention in the documentation about the required version of Fabric.

**Comment by csansone** [comment]
I have the same issue on Python 3.4.2 and Fabric 2.1.3

```
Traceback (most recent call last):
  File "/home/chris/.virtualenvs/pelican3/bin/fab", line 11, in <module>
    sys.exit(program.run())
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/site-packages/invoke/program.py", line 321, in run
    self.parse_collection()
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/site-packages/invoke/program.py", line 399, in parse_collection
    self.load_collection()
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/site-packages/fabric/main.py", line 75, in load_collection
    super(Fab, self).load_collection()
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/site-packages/invoke/program.py", line 588, in load_collection
    module, parent = loader.load(coll_name)
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/site-packages/invoke/loader.py", line 75, in load
    module = imp.load_module(name, fd, path, desc)
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/imp.py", line 235, in load_module
    return load_source(name, filename, file)
  File "/home/chris/.virtualenvs/pelican3/lib/python3.4/imp.py", line 171, in load_source
    module = methods.load()
  File "<frozen importlib._bootstrap>", line 1220, in load
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/home/chris/projects/pelican3/fabfile.py", line 1, in <module>
    from fabric.api import *
ImportError: No module named 'fabric.api'
```

**Comment by avaris** [comment]
It's not that we don't test it, but `fabric` until very recently was a `Python 2`-only library. So, code inside `fabfile.py` reflected that.

Having said that, this is already fixed in master (See #2184). Just not in any release yet.

There is also #2357 to update `fabric` to the new restructured `invoke`.

**Comment by justinmayer** [comment]
The `No module named 'SocketServer'` error will be fixed by (hopefully) releasing Pelican 3.8 in the near future, and as @avaris said, there's a pull request open to transition from Fabric to Invoke.
